app/deployment/remote_slave_service.py

```diff
@@ -20,6 +22,8 @@ def start(self, master_host, master_port, slave_port, num_executors):
         :param num_executors: the number of concurrent executors that will run in this slave service
         :type num_executors: int
         """
+        if master_host == 'localhost':
+            master_host = socket.gethostname()
```

Comment:
Ah, nice. I wonder if the 'localhost' ambiguity could bite us anywhere else / if it would be useful to do this translation at a higher level in the app. (I wouldn't expect that in this PR but I am just thinking out loud. :smile:)